kolibri/content/serializers.py

```diff
@@ -261,6 +263,44 @@ class Meta:
         list_serializer_class = ContentNodeListSerializer
 
 
+class ContentNodeGranularSerializer(serializers.ModelSerializer):
+    total_resources = serializers.SerializerMethodField()
+    resources_on_device = serializers.SerializerMethodField()
+    importable = serializers.SerializerMethodField()
+
+    def get_total_resources(self, obj):
+        total_resources = obj.get_descendants(include_self=True).exclude(kind=content_kinds.TOPIC).count()
+
+        return total_resources
+
+    def get_resources_on_device(self, obj):
+        available_resources = obj.get_descendants(include_self=True).exclude(kind=content_kinds.TOPIC).filter(available=True).count()
+
+        return available_resources
+
+    def get_importable(self, obj):
+        if 'request' not in self.context or not self.context['request'].query_params.get('datafolder', None) or obj.kind == content_kinds.TOPIC:
+            return True
+        else:
+            datafolder = self.context['request'].query_params.get('datafolder', None)
+            files = FileSerializer(obj.files.all(), many=True, read_only=True).data
+            if not files:
+                return False
+
+            importable = True
+            for f in files:
+                # If one of the files under the node is unavailable on the external drive, mark the node as unimportable
+                file_path = get_content_storage_file_path(f['download_url'].split('/')[2], datafolder)
```

Comment:
This code is not very clean.

Would be better to use the filename directly from the LocalFile object, if you are using the model instead, you can do the following:

`file.localfile.get_filename()` see https://github.com/learningequality/kolibri/blob/develop/kolibri/content/models.py#L237

If this is too verbose, you could add the method to the `File` model, to allow direct access on the file object itself.

Either way, this will avoid assumptions needed to split up a download_url just to get the filename.

Reply:
Thank you! I thought LocalFile objects stand for files only exist in the local database so I wouldn't be able to access the localfile name when the file is only in the external drive, but I think my understanding is not correct and will change the code now :)